docs/dev/convert_layout.rst

```diff
@@ -218,24 +224,49 @@ Second example is for a lightly-layout sensitive operator - batch normalization.
 
 ConvertLayout pass is extremely easy to use. The pass is not a part of default relay.build pipeline. The intended usage is to call it between the framework-to-relay parser and relay.build module call.
 
+In order to specify the layouts to convert to, we create a mapping of heavily-layout sensitive operators to a list of the desired layouts for that operator.
+
 .. code-block:: python
 
     # TFlite framework to Relay parser - Default layout is NHWC
     mod, params = relay.frontend.from_tflite(tflite_model,
                                              shape_dict=shape_dict,
                                              dtype_dict=dtype_dict)
 
+    # We assume our model's heavily-layout sensitive operators only consist of nn.conv2d
+    desired_layouts = {'nn.conv2d': ['NCHW']}
```

Comment:
How about having stricter requirement here?

User must specify default
~~~
desired_layouts = {'nn.conv2d': ['NCHW', 'default']}
~~~

@comaniac What do you think? We can add a comment in the Conv2d convert layout function to specify what default means.

Reply:
I agree with you that enforcing default would make the code clearer and actually I personally like this style more. In this case we can also refine the implementation as follows (take conv2d as an example):

```
assert len(desired_layouts) == 2
desired_data_layout, desired_kernel_layout = desired_layouts
```

Reply:
In fact, if this is a case, we can even consider using tuple instead of list to be more type specific (`List[str]` vs. `Tuple[str, str]`).

Reply:
Super like List Conversion to Tuple.

@lhutton1  What do you think? Are you ok with the changes? Thanks for being patient.

Reply:
It's fine to me tho. What I care is whether we can have the following function signature, for example:

```
def convert_qnn_conv2d(desired_layouts: Tuple[str, str])
def convert_qnn_pool(desired_layouts: Tuple[str])
```

But this is a miner point so feel free to merge this PR without it.

Reply:
I see. That makes sense.

For this PR, I think we can start with a dict from python that is Map<String, Array<String>> for now. We can then send a separate PR for stricter type check.

Reply:
I'm happy with either, although Tuple might make slightly more sense. I think it might be easier to do this as a separate PR though.